flowjax/train/variational_fit.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional, Callable
+
+from jax import random
+import optax
+import equinox as eqx
+from tqdm import tqdm
+
+from flowjax.utils import Array
+from flowjax.distributions import Distribution
+
+# Variational targets accept an array of proposed samples and return an array of (usually)
+# the unormalised log posterior probability of the samples.
+VariationalTarget = Callable[[Array], Array]
```

Comment:
I think maybe using the explicit type is better, otherwise `VariationalTarget` may show in the the documentation, and someone may think this is a class called `VariationalTarget`, rather than a type alias

Reply:
My reason for using a type alias here is mainly because the type annotations become quite complex.
e.g for the loss function `Callable[[Distribution, Callable[[Array], Array], random.KeyArray], float]`.
Feels simpler to parse 
```
VariationalLoss = Callable[
    [Distribution, VariationalTarget, random.KeyArray], 
    float
]
```

I'm not super experienced with python typing, so not sure what best practice is here, one alternative I can think of is to explicitly annotate these as `typing.TypeAlias` and move them into a `flowjax.types` submodule? Hopefully that would alleviate confusion.

Not a massive deal, so if you're not keen I will strip these out.

Reply:
I'm not super familiar with what the best practice is either. My inclination would be to shorten it by having it e.g. Callable[Distribution, Callable, random.KeyArray], and just document the argument well enough that it's clear that the second argument maps a sample to an (unnormalised) probability. I agree it's not particularly important so you can leave it if you like (it shouldn't break anything if we want to change it later anyway).